exercises/error-handling/.meta/hints.md

```diff
@@ -0,0 +1,11 @@
+This exercise requires you to handle exceptions. An [exception](https://docs.oracle.com/javase/tutorial/essential/exceptions/definition.html) is an event, which occurs during the execution of a program, that disrupts the normal flow of the program's instructions.
+
+In Java, there are two types of exceptions: checked and unchecked exceptions.
+
+Checked exceptions are the exceptions that are checked at compile time.
```

Comment:
Maybe you could expand a bit on this since it might not be obvious to a beginner what "checked at compile time" means? 🙂 I think it's important to make clear:
1. The practical implications of a checked exception, i.e. that you have to declare it in the method signature of any method that can throw a checked exception, and handle or rethrow it when calling any method that can call a checked exception.
2. That this is because checked exceptions are meant to be handled at runtime, i.e. they are errors you can recover from.
3. Examples of where they are used. They're often used when a method can't return any valid result, for example a search method which hasn't found the item it was searching for. It's an alternative to returning `null` or an error code. A checked exception is better than those alternatives because it forces the user of the method to consider the error case.

Does that make sense? 🙂

Reply:
OK 🙂